crud/delete.lua

```diff
@@ -51,10 +54,58 @@ local function call_delete_on_router(space_name, key, opts)
         return nil, DeleteError:new("Space %q doesn't exist", space_name), true
     end
 
+    local primary_index = space.index[0]
+    local space_format = space:format()
+
     if box.tuple.is(key) then
         key = key:totable()
     end
 
+    local ddl_sharding_key = sharding.get_ddl_sharding_key(space_name)
+    if ddl_sharding_key ~= nil then
+        local sharding_key_fieldno_map, err = utils.get_keys_fieldno_map(space_format, ddl_sharding_key)
+        if next(sharding_key_fieldno_map) == nil then
```

Comment:
Why don't you check `if err ~= nil`?

`next` can't be jitted (http://wiki.luajit.org/NYI). It's better to avoid such instructions if possible. Here it's possible as I see.

Reply:
returned value changed to `nil` in `get_keys_fieldno_map`